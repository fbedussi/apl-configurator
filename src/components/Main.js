import React from 'react';
import { connect } from 'react-redux';

import {
    getSteps, getStepsLeft, getLabels, getCurrentNode,
    getGoingBack, getRequestStatus
} from '../selectors';

import Answer from './Answer';
import Question from './Question';

const mapStateToProps = (state) => ({
    labels: getLabels(state),
    steps: getSteps(state),
    stepsLeft: getStepsLeft(state),
    requestStatus: getRequestStatus(state),
    goingBack: getGoingBack(state),
    currentNode: getCurrentNode(state)
});

const mapDispatchToProps = () => ({
});

export class Main extends React.Component {
    render() {
        if (this.props.requestStatus !== 'unsent') {
            return null;
        }
        const { goingBack, labels, steps, stepsLeft, currentNode } = this.props;

        const className = `workarea ${goingBack ? 'back' : ''}`;

        return (
            <div className="wrapper main-wrapper">
                <div className="progressWrapper">
                    <span className="progressLabel">{labels.configurator}</span>
                    <progress
                        className="progressBar"
                        value={steps - stepsLeft}
                        max={steps}>
                    </progress>
                </div>
                <div
                    className={className}
                >
                    <div className="slide" key={currentNode.id}>
                        {currentNode.questionId && <Question />}
                        {currentNode.answerId && <Answer />}
                    </div>
                </div>
            </div>
        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Main);
