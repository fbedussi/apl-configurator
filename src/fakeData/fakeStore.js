import texts from './apl-texts.json';

const fakeStore = {
    currentLanguage: texts.languageIsoCode,
    showForm: false,
    steps: 6,
    stepsLeft: 6,
    goingBack: false,
    breadcrumbs: [],
    answersHistory: [],
    labels: texts.labels,
    answers: texts.answers,
    questions: texts.questions,
    tree: texts.tree,
    currentNode: texts.tree,
    requestStatus: 'unsent' // || sent || success || error
};

export default fakeStore;
