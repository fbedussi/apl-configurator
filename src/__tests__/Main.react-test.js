import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import Main from '../components/Main';
import renderer from 'react-test-renderer';
import fakeStore from '../fakeData/fakeStore';

const store = createStore(() => fakeStore);

test('Main component - slide', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Main />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Main component - sent', () => {
  store.requestStatus = 'sent'

  const component = renderer.create(
    <Provider store={store}>
      <Main />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Main component - success', () => {
  store.requestStatus = 'success'

  const component = renderer.create(
    <Provider store={store}>
      <Main />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Main component - error', () => {
  store.requestStatus = 'error'

  const component = renderer.create(
    <Provider store={store}>
      <Main />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});