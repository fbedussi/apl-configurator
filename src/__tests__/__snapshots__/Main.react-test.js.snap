// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component - error 1`] = `
<div
  className="wrapper main-wrapper"
>
  <div
    className="progressWrapper"
  >
    <span
      className="progressLabel"
    >
      Configuratore
    </span>
    <progress
      className="progressBar"
      max={6}
      value={0}
    />
  </div>
  <div
    className="workarea "
  >
    <div
      className="slide"
    >
      <div
        className="slideInner question"
      >
        <p
          className="text question-text"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Nel luogo di installazione è possibile collegare il sistema APL alla rete elettrica?",
            }
          }
        />
        <div
          className="question-options"
        >
          <button
            className="question-optionBtn ctaBtn"
            id="1_0"
            onClick={[Function]}
          >
            no
          </button>
          <button
            className="question-optionBtn ctaBtn"
            id="1_1"
            onClick={[Function]}
          >
            sì
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Main component - sent 1`] = `
<div
  className="wrapper main-wrapper"
>
  <div
    className="progressWrapper"
  >
    <span
      className="progressLabel"
    >
      Configuratore
    </span>
    <progress
      className="progressBar"
      max={6}
      value={0}
    />
  </div>
  <div
    className="workarea "
  >
    <div
      className="slide"
    >
      <div
        className="slideInner question"
      >
        <p
          className="text question-text"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Nel luogo di installazione è possibile collegare il sistema APL alla rete elettrica?",
            }
          }
        />
        <div
          className="question-options"
        >
          <button
            className="question-optionBtn ctaBtn"
            id="1_0"
            onClick={[Function]}
          >
            no
          </button>
          <button
            className="question-optionBtn ctaBtn"
            id="1_1"
            onClick={[Function]}
          >
            sì
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Main component - slide 1`] = `
<div
  className="wrapper main-wrapper"
>
  <div
    className="progressWrapper"
  >
    <span
      className="progressLabel"
    >
      Configuratore
    </span>
    <progress
      className="progressBar"
      max={6}
      value={0}
    />
  </div>
  <div
    className="workarea "
  >
    <div
      className="slide"
    >
      <div
        className="slideInner question"
      >
        <p
          className="text question-text"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Nel luogo di installazione è possibile collegare il sistema APL alla rete elettrica?",
            }
          }
        />
        <div
          className="question-options"
        >
          <button
            className="question-optionBtn ctaBtn"
            id="1_0"
            onClick={[Function]}
          >
            no
          </button>
          <button
            className="question-optionBtn ctaBtn"
            id="1_1"
            onClick={[Function]}
          >
            sì
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Main component - success 1`] = `
<div
  className="wrapper main-wrapper"
>
  <div
    className="progressWrapper"
  >
    <span
      className="progressLabel"
    >
      Configuratore
    </span>
    <progress
      className="progressBar"
      max={6}
      value={0}
    />
  </div>
  <div
    className="workarea "
  >
    <div
      className="slide"
    >
      <div
        className="slideInner question"
      >
        <p
          className="text question-text"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Nel luogo di installazione è possibile collegare il sistema APL alla rete elettrica?",
            }
          }
        />
        <div
          className="question-options"
        >
          <button
            className="question-optionBtn ctaBtn"
            id="1_0"
            onClick={[Function]}
          >
            no
          </button>
          <button
            className="question-optionBtn ctaBtn"
            id="1_1"
            onClick={[Function]}
          >
            sì
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
